app/src/lib/git/status.ts

```diff
@@ -75,9 +67,28 @@ export async function getStatus(repository: Repository): Promise<IStatusResult>
   for (const entry of parsePorcelainStatus(result.stdout)) {
     if (entry.kind === 'entry') {
       const status = mapStatus(entry.statusCode)
+
+      if (status.kind === 'ordinary') {
+        if (status.staged === GitStatusEntry.Added
+            && status.unstaged === GitStatusEntry.Deleted) {
+          // we can safely avoid drawing this file in the app
+          continue
+        }
+      }
+
+      if (status.kind === 'untracked') {
+        const existingEntry = files.findIndex(p => p.path === entry.path)
+        if (existingEntry > -1) {
+          // a previous entry for this path exists, remove it
```

Comment:
`git status` will list the untracked files after all the tracked files. This relates to the point above about the "staged delete and untracked file with same path" case.